icetea_lib/GenericProcess.py

```diff
@@ -101,12 +102,12 @@ def _read_fd(file_descr):
                 stream_desc.has_error = True
                 if stream_desc.callback is not None:
                     stream_desc.callback()
-            return
+            return 0
 
         if line:
             stream_desc = NonBlockingStreamReader._get_sd(file_descr)
             if stream_desc is None:
-                return # Process closing
+                return 0 # Process closing
```

Comment:
You do realize that this changes the return value, which might not be what is wanted? If the code that calls this has some error handling bound to return value None this will break it.

Reply:
But this function did not have a return value before, I don't think you could have built any logic on that?

Reply:
A plain return returns None.

Reply:
`GenericProcess::_read_fd()` is "private" method for GenericProcess. Within the GenericProcess, it is called in 3 places.
```
jaakor01@vmxub:~/workspace/icetea
$ grepc -Ir _read_fd
icetea_lib/GenericProcess.py:89:    def _read_fd(file_descr):
icetea_lib/GenericProcess.py:146:                    NonBlockingStreamReader._read_fd(file_descr)
icetea_lib/GenericProcess.py:150:                    while NonBlockingStreamReader._read_fd(file_descr) != 0:
icetea_lib/GenericProcess.py:182:                    NonBlockingStreamReader._read_fd(event.ident)
```

The return value check in lineno 150 is actually part of this PR change.

Also as shown above the grep was done for the whole project with no matches to calls for _read_fd().

Reply:
Ok.